frontend/src/component/project/Project/ProjectFeatureToggles/ProjectFeatureToggles.tsx

```diff
@@ -127,6 +135,16 @@ export const ProjectFeatureToggles = ({
         changeRequestDialogDetails,
     } = useChangeRequestToggle(projectId);
 
+    const hasOtherEnvsWithVariants = (featureName: string) => {
+        const featureEnvs =
+            features.find(feature => feature.name === featureName)
+                ?.environments || [];
+        const found = featureEnvs.find(
+            env => env.enabled && env.variantCount > 0
+        );
+        return found !== undefined;
+    };
+
```

Comment:
@nunogois I'd like to get your advice on this piece of code. I don't like it but I'm limited in my React knowledge.

Reply:
Hmm on a first look I see it could be simplified as:

```suggestion
    const hasOtherEnvsWithVariants = (featureName: string) => {
        const featureEnvs =
            features.find(feature => feature.name === featureName)
                ?.environments || [];
        return featureEnvs.some(env => env.enabled && env.variantCount > 0);
    };

```

Or take it one step further as:

```suggestion
    const hasOtherEnvsWithVariants = (featureName: string) =>
      features.some(
          feature =>
              feature.name === featureName &&
              feature.environments.some(
                  env => env.enabled && env.variantCount > 0
              )
      );

```

Reply:
Yes, but I wanted to keep this logic closer to [this line](https://github.com/Unleash/unleash/pull/2948/files#diff-a1c10fdcd539d988f74d4c3742cc6b8844152999a6c7e509ed8b1765c604524fR305) because here I'm using global `features` variable to find a feature. But at line 305 I already have the feature. 

The problem I faced is that the shape of that feature is different than the shape of the feature in this `features` array, in particular, it's missing access to other environments (the type of the `environments` is different IIRC). If I could do that, then I wouldn't have to iterate on features to find the feature, because I'd already have it